File: CfxGenerator/AssemblyResources.vb
Imports System.Reflection

Public Class AssemblyResources

    Private Shared m_resourceNames As String()
    Public Shared ReadOnly Property ResourceNames As String()
        Get
            
            If m_resourceNames Is Nothing Then
                Dim asm = Assembly.GetExecutingAssembly()
                m_resourceNames = asm.GetManifestResourceNames()
            End If
            Return m_resourceNames
        End Get
    End Property

    Public Shared Function GetResourceName(partialName As String) As String
        Dim name As String = Nothing
        For Each rn In ResourceNames
            If rn.Contains(partialName) Then
                Return rn
            End If
        Next
        Return Nothing
    End Function

    Public Shared Function GetStream(partialName As String) As System.IO.Stream
        Dim name = GetResourceName(partialName)
        If name Is Nothing Then Return Nothing
        Return Assembly.GetExecutingAssembly().GetManifestResourceStream(name)
    End Function

    Public Shared Function GetBinary(partialName As String) As Byte()
        Dim stream = GetStream(partialName)
        Dim data = New IO.MemoryStream
        stream.CopyTo(data)
        Return data.ToArray()
    End Function

    Public Shared Function GetString(partialName As String) As String
        Using stream = GetStream(partialName)
            Using reader = New IO.StreamReader(stream)
                Return reader.ReadToEnd()
            End Using
        End Using
    End Function

    Public Shared Function GetLines(partialName As String) As String()
        Using stream = GetStream(partialName)
            Using reader = New IO.StreamReader(stream)
                Dim ll = New List(Of String)
                Dim l = reader.ReadLine()
                While l IsNot Nothing
                    ll.Add(l)
                    l = reader.ReadLine()
                End While
                Return ll.ToArray()
            End Using
        End Using
    End Function

    Public Shared Function GetData(partialName As String) As Byte()
        Dim data = New System.IO.MemoryStream
        Using stream = GetStream(partialName)
            stream.CopyTo(data)
        End Using
        Return data.GetBuffer()
    End Function

End Class

File: CfxGenerator/AssemblyResources.cs
// Copyright (c) 2014-2017 Wolfgang Borgsmüller
// All rights reserved.
// 
// This software may be modified and distributed under the terms
// of the BSD license. See the License.txt file for details.

using System.Collections.Generic;
using System.Reflection;

public class AssemblyResources {
    private static string[] m_resourceNames;

    public static string[] ResourceNames {
        get {
            if(m_resourceNames == null) {
                var asm = Assembly.GetExecutingAssembly();
                m_resourceNames = asm.GetManifestResourceNames();
            }
            return m_resourceNames;
        }
    }

    public static string GetResourceName(string partialName) {
        foreach(var rn in ResourceNames) {
            if(rn.Contains(partialName)) {
                return rn;
            }
        }
        return null;
    }

    public static System.IO.Stream GetStream(string partialName) {
        var name = GetResourceName(partialName);
        if(name == null)
            return null;
        return Assembly.GetExecutingAssembly().GetManifestResourceStream(name);
    }

    public static byte[] GetBinary(string partialName) {
        var stream = GetStream(partialName);
        var data = new System.IO.MemoryStream();
        stream.CopyTo(data);
        return data.ToArray();
    }

    public static string GetString(string partialName) {
        using(var stream = GetStream(partialName)) {
            return new System.IO.StreamReader(stream).ReadToEnd();
        }
    }

    public static string[] GetLines(string partialName) {
        using(var stream = GetStream(partialName)) {
            var reader = new System.IO.StreamReader(stream);
            var ll = new List<string>();
            var l = reader.ReadLine();
            while(l != null) {
                ll.Add(l);
                l = reader.ReadLine();
            }
            return ll.ToArray();
        }
    }

    public static byte[] GetData(string partialName) {
        var data = new System.IO.MemoryStream();
        using(var stream = GetStream(partialName)) {
            stream.CopyTo(data);
        }
        return data.GetBuffer();
    }
}